Framework/PythonInterface/plugins/algorithms/FindMultipleUMatrices.py

```diff
@@ -0,0 +1,380 @@
+    DataProcessorAlgorithm,
+    AlgorithmFactory,
+    Progress,
+    WorkspaceGroupProperty,
+    IPeaksWorkspaceProperty,
+    AnalysisDataService as ADS,
+)
+from mantid.kernel import Direction, IntBoundedValidator, FloatBoundedValidator, EnabledWhenProperty, PropertyCriterion, logger
+from mantid.geometry import SpaceGroupFactory, PointGroupFactory, CrystalStructure, ReflectionGenerator, ReflectionConditionFilter
+from itertools import combinations, product
+import numpy as np
+
+
+_MIN_NUM_INDEXED_PEAKS = 3  # one more peak indexed than required for CalculateUMatrix
+
+
+class FindMultipleUMatrices(DataProcessorAlgorithm):
+    def name(self):
+        return "FindMultipleUBUsingLatticeParameters"
+
+    def category(self):
+        return "Diffraction\\Reduction;Crystal\\UBMatrix"
+
+    def seeAlso(self):
+        return ["FindUBUsingLatticeParameters", "CalculateUMatrix"]
+
+    def summary(self):
+        return (
+            "Finds multiple UB matrices using lattice parameters - can be used to find UBs in the presence of "
+            "mutiple domains, crystallites or in the presence of spurious peaks."
+        )
+
+    def PyInit(self):
+        # Input
+        self.declareProperty(
+            IPeaksWorkspaceProperty(name="PeaksWorkspace", defaultValue="", direction=Direction.Input),
+            doc="A PeaksWorkspace containing the peaks to integrate.",
+        )
+        self.declareProperty(
+            WorkspaceGroupProperty("OutputWorkspace", "", direction=Direction.Output),
+            doc="The output peak workspaces with UBs set - 1 per UB requested.",
+        )
+        self.declareProperty(
+            name="NumberOfUBs",
+            defaultValue=1,
+            direction=Direction.Input,
+            validator=IntBoundedValidator(lower=1),
+            doc="Number of UB matrices to find.",
+        )
+        positiveFloatValidator = FloatBoundedValidator(lower=0.0)
+        angleValidator = FloatBoundedValidator(lower=0.0, upper=180.0)
+        self.declareProperty(
+            name="a", defaultValue=1.0, direction=Direction.Input, validator=positiveFloatValidator, doc="Lattice parameter a"
+        )
+        self.declareProperty(
+            name="b", defaultValue=1.0, direction=Direction.Input, validator=positiveFloatValidator, doc="Lattice parameter b"
+        )
+        self.declareProperty(
+            name="c", defaultValue=1.0, direction=Direction.Input, validator=positiveFloatValidator, doc="Lattice parameter c"
+        )
+        self.declareProperty(
+            name="alpha", defaultValue=90.0, direction=Direction.Input, validator=angleValidator, doc="Lattice angle alpha"
+        )
+        self.declareProperty(name="beta", defaultValue=90.0, direction=Direction.Input, validator=angleValidator, doc="Lattice angle beta")
+        self.declareProperty(
+            name="gamma", defaultValue=90.0, direction=Direction.Input, validator=angleValidator, doc="Lattice angle gamma"
+        )
```

Comment:
Would it be useful to add the units `(degrees)` in the documentation string so as to be clear its not in radians? And a similar question for some of the other algorithm properties